### PR TITLE
Enhance getting-started documentation

### PR DESCRIPTION
**Pull request by jamtaylo**
Add a note to CONTRIBUTING.md to let the developer know that the git
hooks require additional tools available in the PATH (e.g., rubocop).

Edit README-develop-node-utils-APIs.md to make the getting-start portion
of the docs a little more explicit for developers who may be just
starting out with git and/or ruby.  Reorganized a few paragraphs to more
clearly separate the things that are one-time setup vs. things that are
likely to be done on each feature branch.  Added a paragraph about
installing the project's git hooks.

### CODE REVIEW

File: docs/README-develop-node-utils-APIs.md [hunk 1]
```diff
@@ -69,10 +100,7 @@ git branch
   feature/eigrp
 ```
 
-
-## <a name="complex">Example: router eigrp</a>
-
-Before you start working on the eigrp feature, checkout the feature branch you created earlier.
+Before you start working on the eigrp feature, checkout the feature branch.
```
Comment (robert-w-gries) [inline docs/README-develop-node-utils-APIs.md]: Nit for clarity:

``` diff
-Before you start working on the eigrp feature, checkout the feature branch.
+Before you start working on the eigrp feature, checkout the `feature/eigrp` branch.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Cisco Network Elements support a rich set of features to make networks robust, e
 ## Making Changes
 
 * Fork and clone the repository
-  * Run the `bin/git/update-hooks` script to install our recommended Git hooks into your local repository.
+  * Run the `bin/git/update-hooks` script to install our recommended Git hooks into your local repository.  (Note: some of the hooks require additional tools, such as the `rubocop` ruby gem, be installed and available in your `$PATH`.)
```
Comment (robert-w-gries) [inline CONTRIBUTING.md]: Adding a pointer to your [Ruby Prerequisites](https://github.com/jamtaylo/cisco-network-node-utils/blob/getting-started-docs/docs/README-develop-node-utils-APIs.md#ruby-prerequisites) might be helpful.  I'm not sure what additional tools are needed other than rubocop.

**Comment by robert-w-gries** [comment]
:+1: Thanks for helping improve the docs!

**Comment by jamtaylo** [comment]
Updated based on feedback from @robert-w-gries

File: docs/README-develop-node-utils-APIs.md [hunk 3]
```diff
@@ -33,7 +45,7 @@ gem install simplecov
 gem install minitest
 ```
 
-**NOTE:** If you are working from a server where you don't have admin/root privilages, use the following commands to install the gems and then update the `PATH` to include `~/.gem/ruby/x.x.x/bin`
+**NOTE:** If you are working from a server where you don't have admin/root privilages, use the following commands to install the gems
```
Comment (chrisvanheuveln) [inline docs/README-develop-node-utils-APIs.md]: `privileges`

**Comment by chrisvanheuveln** [comment]
:+1:

File: docs/README-develop-node-utils-APIs.md (hunk 3 above)
Comment (jamtaylo) [inline docs/README-develop-node-utils-APIs.md]: fixed in 3c6dc79

**Pull request merged**